### PR TITLE
[MINOR][CORE] When failed to canceling the job group, add a warning log

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to add add a warning log when failed to canceling the job group.

### Why are the changes needed?
In production, we found that when there are many jobs, if we specify `a just-submitted job group` and `cancel` it using the API `SparkContext#cancelJobGroup(groupId: String)`, the cancellation will fail (`silently`) because the job has not yet entered `activeJobs`, and there is `no warning` prompt.
(PS: Due to historical reasons, we are not currently using the newer API `SparkContext#cancelJobGroupAndFutureJobs(groupId: String)`)

I propose to add `a warning log` here to help troubleshoot `issues`.

### Does this PR introduce _any_ user-facing change?
Yes, only for log.

### How was this patch tested?
N/A

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @cloud-fan @HyukjinKwon

File: core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala [hunk 1]
```diff
@@ -1220,6 +1220,10 @@ private[spark] class DAGScheduler(
         _.getProperty(SparkContext.SPARK_JOB_GROUP_ID) == groupId
       }
     }
+    if (activeInGroup.isEmpty && !cancelFutureJobs) {
+      logWarning(s"Cancel job group $groupId failed due to " +
```
Comment (cloud-fan) [inline core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala:1224]: `Failed to cancel job group $groupId. Cannot find active jobs for it.`

**Review by cloud-fan — APPROVED** [review]

File: core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala (hunk 1 above)
Comment (panbingkun) [reply core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala]: Done.
Thanks!

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Merged to master. Thank you all.

**Pull request closed without merging**